crates/transaction-emitter-lib/src/transaction_generator/nft_mint_and_transfer.rs

```diff
@@ -31,27 +38,52 @@ impl NFTMint {
             creator_account,
             collection_name,
             token_name,
+            nft_balance: HashMap::new(),
         }
     }
 }
 
-impl TransactionGenerator for NFTMint {
+impl TransactionGenerator for NFTMintAndTransfer {
     fn generate_transactions(
         &mut self,
         accounts: Vec<&mut LocalAccount>,
         transactions_per_account: usize,
     ) -> Vec<SignedTransaction> {
         let mut requests = Vec::with_capacity(accounts.len() * transactions_per_account);
+        let mut visited_accounts: Vec<&LocalAccount> = vec![];
         for account in accounts {
             for _ in 0..transactions_per_account {
-                requests.push(create_nft_transfer_request(
-                    account,
-                    &self.creator_account,
-                    &self.collection_name,
-                    &self.token_name,
-                    &self.txn_factory,
-                ));
+                requests.push({
+                    let balance = *self.nft_balance.get(&account.address()).unwrap_or(&0);
+                    if visited_accounts.is_empty() || balance == 0 {
+                        let creator_account = &mut self.creator_account.lock();
```

Comment:
unless this is called very very rarely, this is going to be a bottleneck on the concurrency, as sequence numbers will need to be executed in order, and different threads submit to different clients, etc.

Should we be sending only 1 nft token (instead of full balance) below, and initialize all accounts to have a large number of NFTs to start with?

Reply:
The motivation is to transfer all NFTs from a non-faucet account to another to test NFT transfer traffic and more importantly, TRIGGERing token deletion code in move and storage... if we mint 1M and send 1 by 1 to everywhere we'll never trigger that. Let me see how can I improve concurrency based on that.